From 3987cf1e9203c7ca0a0c34845244864d69ab138a Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Thu, 4 May 2023 11:34:19 +0200
Subject: [PATCH 1/9] Configure network in early phase

We need be able to let the cloud proviers configure the network before
binging it up. Add a `set_network_config` action that is run in early
phase.

Default to simply call assemble-interfaces.
---
 lib/tiny-cloud/init | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/lib/tiny-cloud/init b/lib/tiny-cloud/init
index cf51e98..2294c28 100644
--- a/lib/tiny-cloud/init
+++ b/lib/tiny-cloud/init
@@ -13,6 +13,7 @@
 INIT_ACTIONS_EARLY="
     expand_root
     install_hotplugs
+    set_network_config
 "
 INIT_ACTIONS_MAIN="
     save_userdata
@@ -73,6 +74,10 @@ init__install_hotplugs() {
     return $rc
 }
 
+init__set_network_config() {
+    assemble-interfaces
+}
+
 ### init-main functions
 
 init__set_hostname() {
-- 
2.40.1

